test/src/test/java/hudson/tasks/LogRotatorPeriodicTaskTest.java

```diff
@@ -0,0 +1,409 @@
+import hudson.model.LogRotatorMapping;
+import hudson.model.TaskListener;
+import hudson.util.StreamTaskListener;
+import jenkins.model.GlobalConfiguration;
+
+public class LogRotatorPeriodicTaskTest {
+    
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+    
+    protected LogRotator mockCustomRotator1;
+    
+    protected LogRotator mockCustomRotator2;
+    
+    /**
+     * A mock log rotator that will only apply some jobs.
+     */
+    protected LogRotator mockSpecificGlobalRotator;
+    
+    /**
+     * Maps the {@link #mockSpecificGlobalRotator} to a certain job name pattern
+     */
+    protected LogRotatorMapping specificGlobalMapping;
+    
+    /**
+     * A mock log rotator that will apply to all jobs.
+     */
+    protected LogRotator mockCatchallGlobalRotator;
+    
+    /**
+     * Maps the {@link #mockCatchallGlobalRotator} to the regex ".*"
+     */
+    protected LogRotatorMapping catchallGlobalMapping;
+    
+    /**
+     * The task under test
+     */
+    protected LogRotatorPeriodicTask task;
+    
+    /**
+     * Actual text written to the listener would go here
+     * (but, {@link LogRotatorPeriodicTask} doesn't write anything to the listener)
+     */
+    protected StringWriter log;
+    
+    /**
+     * TaskListener for the periodic log rotation
+     */
+    protected TaskListener listener;
+    
+    @Before
+    public void setUp() throws IOException {
+        
+        mockCustomRotator1 = mock( LogRotator.class );
+        
+        mockCustomRotator2 = mock( LogRotator.class );
+        
+        mockSpecificGlobalRotator = mock( LogRotator.class );
+        specificGlobalMapping = new LogRotatorMapping( "rotator-.*", mockSpecificGlobalRotator );
+        
+        mockCatchallGlobalRotator = mock( LogRotator.class );
+        catchallGlobalMapping = new LogRotatorMapping( ".*", mockCatchallGlobalRotator );
+        
+        task = new LogRotatorPeriodicTask();
+        
+        log = new StringWriter();
+        
+        listener = new StreamTaskListener( log );
+    }
+    
+    
+    @Test
+    public void customRotators_correctlyIgnored() throws IOException, InterruptedException {
+        
+        /*
+         * Test Setup
+         */
+        
+        // create a project with a LogRotator
+        FreeStyleProject project = j.createFreeStyleProject();
+        project.setBuildDiscarder( mockCustomRotator1 );
+        
+        // set global Log Rotation policy for jobs with a custom rotator:
+        // do nothing
+        LogRotatorConfiguration config = GlobalConfiguration.all().get( LogRotatorConfiguration.class );
+        assertNotNull( config );
+        config.setPolicyForJobsWithCustomLogRotator( LogRotationPolicy.NONE );
+        
+        // this global LogRotator should match all jobs.
+        // but since the policy is "do nothing," it should not be used.
+        config.setGlobalLogRotators(Arrays.asList( new LogRotatorMapping[] {
```

Comment:
I prefer to be explicit; once in a blue moon the compiler is happier with the generic outcome when it's all spelled out unambiguously. It's just a few extra keypresses in an IDE with autocomplete, but it may prevent a pernicious headache.

In this case it seems to be unnecessary; I've updated the test class to omit the array.